network/p2p/scoring/score_option.go

```diff
@@ -136,20 +159,65 @@ func (s *ScoreOption) preparePeerScoreParams() {
 
 		// AppSpecificScore is a function that takes a peer ID and returns an application specific score.
 		// At the current stage, we only use it to penalize and reward the peers based on their subscriptions.
-		AppSpecificScore: func(pid peer.ID) float64 {
-			if err := s.validator.CheckSubscribedToAllowedTopics(pid); err != nil {
-				s.logger.Error().
-					Err(err).
-					Str("peer_id", pid.String()).
-					Msg("invalid subscription detected, penalizing peer")
-				return MaxAppSpecificPenalty
-			}
-			s.logger.Trace().
-				Str("peer_id", pid.String()).
-				Msg("subscribed topics for peer validated, rewarding peer")
-			return MaxAppSpecificReward
-		},
+		AppSpecificScore: s.appSpecificScoreFunction,
 		// AppSpecificWeight is the weight of the application specific score.
 		AppSpecificWeight: DefaultAppSpecificScoreWeight,
 	}
 }
+
+func (s *ScoreOption) BuildGossipSubScoreOption() pubsub.Option {
+	s.preparePeerScoreParams()
+	s.preparePeerScoreThresholds()
+
+	s.logger.Info().
+		Float64("gossip_threshold", s.peerThresholdParams.GossipThreshold).
+		Float64("publish_threshold", s.peerThresholdParams.PublishThreshold).
+		Float64("graylist_threshold", s.peerThresholdParams.GraylistThreshold).
+		Float64("accept_px_threshold", s.peerThresholdParams.AcceptPXThreshold).
+		Float64("opportunistic_graft_threshold", s.peerThresholdParams.OpportunisticGraftThreshold).
+		Msg("peer score thresholds configured")
+
+	return pubsub.WithPeerScore(
+		s.peerScoreParams,
+		s.peerThresholdParams,
+	)
+}
+
+func defaultAppSpecificScoreFunction(logger zerolog.Logger, idProvider module.IdentityProvider, validator *SubscriptionValidator) func(peer.ID) float64 {
+	return func(pid peer.ID) float64 {
+		lg := logger.With().Str("peer_id", pid.String()).Logger()
+
+		// checks if peer has a valid Flow protocol identity.
+		flowId, err := HasValidFlowIdentity(idProvider, pid)
+		if err != nil {
+			lg.Error().
+				Err(err).
+				Msg("invalid peer identity, penalizing peer")
+			return MaxAppSpecificPenalty
+		}
+
+		lg = lg.With().
+			Hex("flow_id", logging.ID(flowId.NodeID)).
+			Str("role", flowId.Role.String()).
+			Logger()
+
+		// checks if peer has any subscription violation.
+		if err := validator.CheckSubscribedToAllowedTopics(pid, flowId.Role); err != nil {
+			lg.Err(err).
+				Msg("invalid subscription detected, penalizing peer")
+			return MaxAppSpecificPenalty
+		}
+
+		// checks if peer is an access node, and if so, pushes it to the
+		// edges of the network by giving the minimum penalty.
+		if flowId.Role == flow.RoleAccess {
+			lg.Info().
```

Comment:
how often is this logged? is `Debug` level more appropriate?

Reply:
https://github.com/onflow/flow-go/pull/3379/commits/3659b07a53e233a4fd07e277872dfc9d143385d1